### PR TITLE
chore: polish various license content

### PR DESCRIPTION
**Pull request by tisonkun**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

This is a follow-up to https://github.com/datafuselabs/databend/pull/11153.

Please review one commit by another.

### CODE REVIEW

File: licenses/Apache-2.0.txt [hunk 1]
```diff
@@ -186,7 +186,7 @@
       same "printed page" as the copyright notice for easier
       identification within third-party archives.
 
-   Copyright 2021 Datafuse Labs
+   Copyright [yyyy] [name of copyright owner]
```
Comment (tisonkun) [inline licenses/Apache-2.0.txt:189]: The ALv2 license is immutable AS IS. You don't need to "fill in the blank".

File: src/binaries/meta-upgrade-09/src/rewrite.rs [hunk 2]
```diff
@@ -1,4 +1,4 @@
-// Copyright 2023 Datafuse Labs.
+// Copyright 2021 Datafuse Labs
```
Comment (tisonkun) [inline src/binaries/meta-upgrade-09/src/rewrite.rs:1]: See https://issues.apache.org/jira/browse/LEGAL-629 for more details. Here we use the interception year of the project.

File: licenserc.toml [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+    "**/*.yaml",
+    "**/*.toml",
+    "**/*.lock",
+    "**/*.yapf",
+    "**/*.test",
+    "**/*.txt",
+]
+
+[properties]
+inceptionYear = 2021
```
Comment (ZhiHanZ) [inline licenserc.toml:55]: We could update license header year and inceptor year to 2023

**Comment by ZhiHanZ** [comment]
LGTM, thanks @tisonkun

**Review by ZhiHanZ — APPROVED** [review]

**Review by PsiACE — APPROVED** [review]
lgtm, thanks

**Comment by BohuTANG** [comment]
Thanks @tisonkun 💯

**Comment by tisonkun** [comment]
Feedback - this PR is somehow better to be "rebase and merge", while current status is also _OK_.

**Pull request merged**